### PR TITLE
MH3: 3 new Flipwalkers

### PR DESCRIPTION
**Pull request by MikeS-NZ**
MH3: Add card scripts for the last three leaked flipwalker cards:

- Ral, Monsoon Mage // Ral, Leyline Prodigy
- Sorin of House Markov // Sorin, Ravenous Neonate
- Tamiyo, Inquisitive Student // Tamiyo, Seasoned Scholar

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/ral_monsoon_mage_ral_leyline_prodigy.txt [hunk 1]
```diff
@@ -0,0 +1,35 @@
+Name:Ral, Monsoon Mage
+ManaCost:1 R
+Types:Legendary Creature Human Wizard
+PT:1/3
+S:Mode$ ReduceCost | ValidCard$ Instant,Sorcery | Type$ Spell | Activator$ You | Amount$ 1 | Description$ Instant and sorcery spells you cast cost {1} less to cast.
+T:Mode$ SpellCast | ValidCard$ Instant,Sorcery | ValidActivatingPlayer$ You | TriggerZones$ Battlefield | PlayerTurn$ True | Execute$ TrigFlip | TriggerDescription$ Whenever you cast an instant or sorcery spell during your turn, flip a coin. If you lose the flip, CARDNAME deals 1 damage to you. If you win the flip, you may exile NICKNAME. If you do, return him to the battlefield transformed under his owner's control.
+SVar:TrigFlip:DB$ FlipACoin | WinSubAbility$ DBTransform | LoseSubAbility$ DBDamage
+SVar:DBDamage:DB$ DealDamage | Defined$ You | NumDmg$ 1
+SVar:DBCopy:DB$ CopySpellAbility | Defined$ TriggeredSpellAbility | AILogic$ Always | MayChooseTarget$ True
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/ral_monsoon_mage_ral_leyline_prodigy.txt:9]: ?

File: forge-gui/res/cardsfolder/upcoming/ral_monsoon_mage_ral_leyline_prodigy.txt [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+SVar:DBTransform:DB$ ChangeZone | Origin$ Battlefield | Destination$ Exile | OptionalDecider$ You | RememberChanged$ True | SubAbility$ DBReturn
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/ral_monsoon_mage_ral_leyline_prodigy.txt:10]: OptionalDecider does not exist (here)

File: forge-gui/res/cardsfolder/upcoming/ral_monsoon_mage_ral_leyline_prodigy.txt [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+SVar:DBEffect:DB$ Effect | StaticAbilities$ EffPlay | EffectOwner$ You | RememberObjects$ Remembered | ForgetOnMoved$ Exile | SubAbility$ DBCleanup
+SVar:EffPlay:Mode$ Continuous | MayPlay$ True | MayPlayWithoutManaCost$ True | EffectZone$ Command | Affected$ Card.Instant+IsRemembered,Card.Sorcery+IsRemembered | AffectedZone$ Exile | Description$ You may cast instant and sorcery cards from among the exiled cards this turn without paying their mana costs.
+SVar:DBCleanup:DB$ Cleanup | ClearRemembered$ True
+SVar:X:Count$ThisTurnCast_Instant.YouCtrl,Sorcery.YouCtrl
+DeckNeeds:Type$Instant|Sorcery
+Oracle:[+1]: Until your next turn, instant and sorcery spells you cast cost {1} less to cast.\n[-2]: Ral deals 2 damage divided as you choose among one or two targets. Draw a card if you control a blue permanent other than Ral.\n[-8]: Exile the top eight cards of your library. You may cast instant and sorcery spells from among them this turn without paying their mana costs.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/ral_monsoon_mage_ral_leyline_prodigy.txt:35]: oracle incomplete?

File: forge-gui/res/cardsfolder/upcoming/ral_monsoon_mage_ral_leyline_prodigy.txt [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+Name:Ral, Leyline Prodigy
+ManaCost:no cost
+Colors:blue,red
+Types:Legendary Planeswalker Ral
+Loyalty:2
+K:etbCounter:LOYALTY:X:no Condition:CARDNAME enters the battlefield with an additional loyalty counter on him for each instant and sorcery spell you've cast this turn.
+A:AB$ Effect | Cost$ AddCounter<1/LOYALTY> | Planeswalker$ True | StaticAbilities$ RalReduceCost | Duration$ UntilYourNextTurn | SpellDescription$ Until your next turn, instant and sorcery spells you cast cost {1} less to cast.
+SVar:RalReduceCost:Mode$ ReduceCost | EffectZone$ Command | ValidCard$ Instant,Sorcery | Type$ Spell | Activator$ You | Amount$ 1 | Description$ Until your next turn, instant and sorcery spells you cast cost {1} less to cast.
+A:AB$ DealDamage | Cost$ SubCounter<2/LOYALTY> | Planeswalker$ True | ValidTgts$ Creature,Player,Planeswalker | TgtPrompt$ Select any target to distribute damage to | NumDmg$ 2 | TargetMin$ 1 | TargetMax$ 2 | DividedAsYouChoose$ 2 | SubAbility$ DBDraw | SpellDescription$ NICKNAME deals 2 damage divided as you choose among one or two targets. Draw a card if you control a blue permanent other than NICKNAME.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/ral_monsoon_mage_ral_leyline_prodigy.txt:27]: wrong tgts

File: forge-gui/res/cardsfolder/upcoming/ral_monsoon_mage_ral_leyline_prodigy.txt [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+SVar:DBDraw:DB$ Draw | NumCards$ 1 | ConditionPresent$ Permanent.Blue+YouCtrl+Other
+A:AB$ Dig | Cost$ SubCounter<8/LOYALTY> | Planeswalker$ True | Ultimate$ True | Defined$ You | DigNum$ 8 | ChangeNum$ All | DestinationZone$ Exile | RememberChanged$ True | SubAbility$ DBEffect | SpellDescription$ Exile the top eight cards of your library. You may cast instant and sorcery spells from among them this turn without paying their mana costs.
+SVar:DBEffect:DB$ Effect | StaticAbilities$ EffPlay | EffectOwner$ You | RememberObjects$ Remembered | ForgetOnMoved$ Exile | SubAbility$ DBCleanup
+SVar:EffPlay:Mode$ Continuous | MayPlay$ True | MayPlayWithoutManaCost$ True | EffectZone$ Command | Affected$ Card.Instant+IsRemembered,Card.Sorcery+IsRemembered | AffectedZone$ Exile | Description$ You may cast instant and sorcery cards from among the exiled cards this turn without paying their mana costs.
+SVar:DBCleanup:DB$ Cleanup | ClearRemembered$ True
+SVar:X:Count$ThisTurnCast_Instant.YouCtrl,Sorcery.YouCtrl
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/ral_monsoon_mage_ral_leyline_prodigy.txt:33]: ?

File: forge-gui/res/cardsfolder/upcoming/sorin_of_house_markov_sorin_ravenous_neonate.txt [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+ALTERNATE
+
+Name:Sorin, Ravenous Neonate
+ManaCost:no cost
+Types:Legendary Planeswalker Sorin
+Colors:white,black
+Loyalty:3
+A:AB$ Token | Cost$ AddCounter<2/LOYALTY> | Planeswalker$ True | TokenAmount$ 1 | TokenScript$ c_a_food_sac | TokenOwner$ You | SpellDescription$ Create a Food token.
+A:AB$ DealDamage | Cost$ SubCounter<1/LOYALTY> | Planeswalker$ True | ValidTgts$ Creature,Player,Planeswalker | TgtPrompt$ Select any target | NumDmg$ X | SpellDescription$ CARDNAME deals damage equal to the amount of life you gained this turn to any target.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/sorin_of_house_markov_sorin_ravenous_neonate.txt:25]: wrong tgts

File: forge-gui/res/cardsfolder/upcoming/tamiyo_inquisitive_student_tamiyo_seasoned_scholar.txt [hunk 7]
```diff
@@ -0,0 +1,34 @@
+Name:Tamiyo, Inquisitive Student
+ManaCost:U
+Types:Legendary Creature Moonfolk Wizard
+PT:0/3
+K:Flying
+T:Mode$ Attacks | ValidCard$ Card.Self | Execute$ TrigInvestigate | TriggerZones$ Battlefield | TriggerDescription$ Whenever CARDNAME attacks, investigate. (Create a Clue token. It's an artifact with "{2}, Sacrifice this artifact: Draw a card.")
+SVar:TrigInvestigate:DB$ Investigate
+T:Mode$ Drawn | ValidCard$ Card.YouCtrl | Number$ 3 | TriggerZones$ Battlefield | Execute$ TrigTransform | TriggerDescription$ Whenever you draw your third card in a turn, exile NICKNAME, then return her to the battlefield transformed under her owner's control.
+SVar:TrigTransform:DB$ ChangeZone | Origin$ Battlefield | Destination$ Exile | RememberChanged$ True | SubAbility$ DBReturn
+SVar:DBReturn:DB$ ChangeZone | Defined$ Remembered | Origin$ Exile | Destination$ Battlefield | Transformed$ True | ForgetOtherRemembered$ True | SubAbility$ DBCleanup | StackDescription$ None
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/tamiyo_inquisitive_student_tamiyo_seasoned_scholar.txt:10]: StackDescription...

File: forge-gui/res/cardsfolder/upcoming/tamiyo_inquisitive_student_tamiyo_seasoned_scholar.txt [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+Oracle:Flying\nWhenever Tamiyo, Inquisitive Student attacks, investigate. (Create a Clue token. It's an artifact with "{2}, Sacrifice this artifact: Draw a card.")\nWhenever you draw your third card in a turn, exile Tamiyo, then return her to the battlefield transformed under her owner's control.
+
+ALTERNATE
+
+Name:Tamiyo, Seasoned Scholar
+ManaCost:no cost
+Colors:green,blue
+Types:Legendary Planeswalker Tamiyo
+Loyalty:2
+A:AB$ Effect | Cost$ AddCounter<2/LOYALTY> | Planeswalker$ True | Triggers$ TrigAttack | Duration$ UntilYourNextTurn | Name$ Tamiyo, Seasoned Scholar Effect | AILogic$ Main2 | SpellDescription$ Until your next turn, whenever a creature an opponent controls attacks you or a planeswalker you control, it gets -1/-0 until end of turn.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/tamiyo_inquisitive_student_tamiyo_seasoned_scholar.txt:24]: identical Name$ is outdated

File: forge-gui/res/cardsfolder/upcoming/tamiyo_inquisitive_student_tamiyo_seasoned_scholar.txt [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+SVar:TrigAttack:Mode$ Attacks | ValidCard$ Creature.OppCtrl | Attacked$ You,Planeswalker.YouCtrl |  TriggerZones$ Command | Execute$ TamiyoPump | TriggerDescription$ Until your next turn, whenever a creature an opponent controls attacks, it gets -1/-0 until end of turn.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/tamiyo_inquisitive_student_tamiyo_seasoned_scholar.txt:25]: wrong description

File: forge-gui/res/cardsfolder/upcoming/ral_monsoon_mage_ral_leyline_prodigy.txt (hunk 4 above)
Comment (MikeS-NZ) [reply forge-gui/res/cardsfolder/upcoming/ral_monsoon_mage_ral_leyline_prodigy.txt:27]: I'm unsure what is wrong here. Ral's -2 is worded exactly the same as Electrolyze, from which this ability borrows code.

File: forge-gui/res/cardsfolder/upcoming/ral_monsoon_mage_ral_leyline_prodigy.txt (hunk 5 above)
Comment (MikeS-NZ) [reply forge-gui/res/cardsfolder/upcoming/ral_monsoon_mage_ral_leyline_prodigy.txt:33]: The X variable tracks the number of instants + sorceries the controller has cast this turn, and adds that many additional loyalty counters to Ral as he enters. Seemed to work alright in testing

File: forge-gui/res/cardsfolder/upcoming/ral_monsoon_mage_ral_leyline_prodigy.txt (hunk 4 above)
Comment (MikeS-NZ) [reply forge-gui/res/cardsfolder/upcoming/ral_monsoon_mage_ral_leyline_prodigy.txt:27]: Battle -- I got it now! Scripting in a post-MOM world feels strange now. Fixing

File: forge-gui/res/cardsfolder/upcoming/ral_monsoon_mage_ral_leyline_prodigy.txt [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+PT:1/3
+S:Mode$ ReduceCost | ValidCard$ Instant,Sorcery | Type$ Spell | Activator$ You | Amount$ 1 | Description$ Instant and sorcery spells you cast cost {1} less to cast.
+T:Mode$ SpellCast | ValidCard$ Instant,Sorcery | ValidActivatingPlayer$ You | TriggerZones$ Battlefield | PlayerTurn$ True | Execute$ TrigFlip | TriggerDescription$ Whenever you cast an instant or sorcery spell during your turn, flip a coin. If you lose the flip, CARDNAME deals 1 damage to you. If you win the flip, you may exile NICKNAME. If you do, return him to the battlefield transformed under his owner's control.
+SVar:TrigFlip:DB$ FlipACoin | WinSubAbility$ DBTransform | LoseSubAbility$ DBDamage
+SVar:DBDamage:DB$ DealDamage | Defined$ You | NumDmg$ 1
+SVar:DBCopy:DB$ CopySpellAbility | Defined$ TriggeredSpellAbility | AILogic$ Always | MayChooseTarget$ True
+SVar:DBTransform:DB$ ChangeZone | Origin$ Battlefield | Destination$ Exile | OptionalDecider$ You | RememberChanged$ True | SubAbility$ DBReturn
+SVar:DBReturn:DB$ ChangeZone | Defined$ Remembered | Origin$ Exile | Destination$ Battlefield | Transformed$ True | ForgetOtherRemembered$ True | SubAbility$ DBCleanup | StackDescription$ None
```
[the remark this answers is not in the document]
Comment (MikeS-NZ) [reply forge-gui/res/cardsfolder/upcoming/ral_monsoon_mage_ral_leyline_prodigy.txt]: Fixed!

File: forge-gui/res/cardsfolder/upcoming/ral_monsoon_mage_ral_leyline_prodigy.txt (hunk 3 above)
Comment (MikeS-NZ) [reply forge-gui/res/cardsfolder/upcoming/ral_monsoon_mage_ral_leyline_prodigy.txt]: Added Ral's ETB replacement static

File: forge-gui/res/cardsfolder/upcoming/sorin_of_house_markov_sorin_ravenous_neonate.txt (hunk 6 above)
Comment (MikeS-NZ) [reply forge-gui/res/cardsfolder/upcoming/sorin_of_house_markov_sorin_ravenous_neonate.txt]: Added "Any"

File: forge-gui/res/cardsfolder/upcoming/tamiyo_inquisitive_student_tamiyo_seasoned_scholar.txt (hunk 7 above)
Comment (MikeS-NZ) [reply forge-gui/res/cardsfolder/upcoming/tamiyo_inquisitive_student_tamiyo_seasoned_scholar.txt]: Fixed

File: forge-gui/res/cardsfolder/upcoming/tamiyo_inquisitive_student_tamiyo_seasoned_scholar.txt (hunk 8 above)
Comment (MikeS-NZ) [reply forge-gui/res/cardsfolder/upcoming/tamiyo_inquisitive_student_tamiyo_seasoned_scholar.txt]: Removed Name$ param

File: forge-gui/res/cardsfolder/upcoming/tamiyo_inquisitive_student_tamiyo_seasoned_scholar.txt (hunk 9 above)
Comment (MikeS-NZ) [reply forge-gui/res/cardsfolder/upcoming/tamiyo_inquisitive_student_tamiyo_seasoned_scholar.txt]: Fixed desc

File: forge-gui/res/cardsfolder/upcoming/ral_monsoon_mage_ral_leyline_prodigy.txt (hunk 2 above)
Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/ral_monsoon_mage_ral_leyline_prodigy.txt]: Optional$ True

Comment (MikeS-NZ) [reply forge-gui/res/cardsfolder/upcoming/ral_monsoon_mage_ral_leyline_prodigy.txt]: Thanks! Added Optional$ param

**Review by tool4ever — APPROVED** [review]

**Pull request merged**